### PR TITLE
added new location tests

### PR DESCRIPTION
**Pull request by AledWEvans**

### CODE REVIEW

File: src/test/java/uk/gov/justice/digital/hmpps/vsip/services/clients/PrisonVisitsTestingHelperClient.java [hunk 1]
```diff
@@ -199,6 +202,96 @@ public void deleteApplication(String applicationReference) {
...
+            LocalDateTime sessionStartDateTime,
+            Integer weeklyFrequency,
+            Integer closedCapacity,
+            Integer openCapacity,
+            String locationLevels,
+            String incentive,
+            String category,
+            Boolean disableAllOtherSessionsForSlotAndPrison
+    ) {
+        final var uri = "/test/prison/"+prisonCode+"/add/template";
```
Comment (adaviesMOJ) [inline src/test/java/uk/gov/justice/digital/hmpps/vsip/services/clients/PrisonVisitsTestingHelperClient.java:216]: Small suggestion to edit the URL to the following:

final var uri = "/test/prison/"+prisonCode+"/template/add";

File: src/test/java/uk/gov/justice/digital/hmpps/vsip/services/clients/PrisonVisitsTestingHelperClient.java [hunk 2]
```diff
@@ -199,6 +202,96 @@ public void deleteApplication(String applicationReference) {
...
+            return results;
+        } catch(Exception e) {
+            LOG.error("Error put createSessionTemplate failed :" + uri, e);
+            throw e;
+        }
+    }
+
+    public void deleteSessionTemplate(String sessionTemplateReference) {
+
+        final var uri = "/test/template/" + sessionTemplateReference + "/delete";
```
Comment (adaviesMOJ) [inline src/test/java/uk/gov/justice/digital/hmpps/vsip/services/clients/PrisonVisitsTestingHelperClient.java:254]: Could this be:

        final var uri = "/test/template/" + sessionTemplateReference;

and then change the webClient.put to a webClient.delete ? and have the endpoint on the helper-api be a @ DeleteMapping ?

Comment (AledWEvans) [reply src/test/java/uk/gov/justice/digital/hmpps/vsip/services/clients/PrisonVisitsTestingHelperClient.java:254]: It's not just delete, it also updates the DB and enables and disables things.

File: src/test/java/uk/gov/justice/digital/hmpps/vsip/services/PrisonVisitsTestingHelperService.java [hunk 3]
```diff
@@ -114,6 +115,22 @@ public void changeVisitPrison(String bookingReference, String prisonCode) {
...
+    public void createSessionTemplate(String prisonCode,
+                                      LocalDateTime sessionStartDateTime,
+                                      Integer weeklyFrequency,
+                                      Integer closedCapacity,
+                                      Integer openCapacity,
+                                      String locationLevels,
+                                      String incentive,
+                                      String category,
+                                      Boolean disableAllOtherSessionsForSlotAndPrison) {
+        LOG.debug("Enter changeVisitPrison {1} {2} {3} {4} {5} {6} {7} {8} {9}", prisonCode, sessionStartDateTime,weeklyFrequency,closedCapacity, openCapacity,locationLevels,incentive,category,disableAllOtherSessionsForSlotAndPrison);
```
Comment (dhirajshettymoj) [inline src/test/java/uk/gov/justice/digital/hmpps/vsip/services/PrisonVisitsTestingHelperService.java:127]: The debug format is {} without the numbers.

**Review by dhirajshettymoj — COMMENTED** [review]
minor comment

File: src/test/java/uk/gov/justice/digital/hmpps/vsip/services/PrisonVisitsTestingHelperService.java (hunk 3 above)
Comment (AledWEvans) [reply src/test/java/uk/gov/justice/digital/hmpps/vsip/services/PrisonVisitsTestingHelperService.java]: updated

**Review by dhirajshettymoj — APPROVED** [review]
LGTM

**Pull request merged**